backend/core/src/main/java/org/sonarsource/sonarlint/core/SonarCloudActiveEnvironment.java

```diff
@@ -20,28 +20,70 @@
 package org.sonarsource.sonarlint.core;
 
 import java.net.URI;
+import java.util.Map;
 
-public class SonarCloudActiveEnvironment {
+import static org.apache.commons.lang.StringUtils.removeEnd;
 
-  public static final URI PRODUCTION_URI = URI.create("https://sonarcloud.io");
+public class SonarCloudActiveEnvironment {
 
   public static SonarCloudActiveEnvironment prod() {
-    return new SonarCloudActiveEnvironment(PRODUCTION_URI, URI.create("wss://events-api.sonarcloud.io/"));
+    return new SonarCloudActiveEnvironment();
   }
 
-  private final URI uri;
-  private final URI webSocketsEndpointUri;
+  private final Map<SonarCloudRegion, SonarQubeCloudUris> uris;
+
+  public SonarCloudActiveEnvironment() {
+    var euUris = new SonarQubeCloudUris(SonarCloudRegion.EU.getProductionUri(), SonarCloudRegion.EU.getWebSocketUri());
+    var usUris = new SonarQubeCloudUris(SonarCloudRegion.US.getProductionUri(), SonarCloudRegion.US.getWebSocketUri());
+    this.uris = Map.of(SonarCloudRegion.EU, euUris, SonarCloudRegion.US, usUris);
```

Comment:
Good move to store the data in the enum. As a consequence I think you don't need the `uris` Map anymore, in most places you can rely on the enum